### PR TITLE
[Core] Support alter database

### PR DESCRIPTION
**Pull request by jerry-024**
### Purpose

Support alter database
Linked issue: #4540

### Tests

- RESTCatalogTest. testAlterDatabase
- RESTCatalogTest. testAlterDatabaseWhenDatabaseNotExistAndIgnoreIfNotExistsIsFalse
- RESTCatalogTest. testAlterDatabaseWhenDatabaseNotExistAndIgnoreIfNotExistsIsTrue
- HiveCatalogTest. testAlterDatabase
- JdbcCatalogTest.testAlterDatabase
- CachingCatalogTest. testInvalidateWhenDatabaseIsAltered

### API and Format

- add alter dabatase API in catalog

### CODE REVIEW

**Review by JingsongLi — COMMENTED** [review]
Can you also support ALTER DATABASE for Flink SQL and Spark SQL?

File: docs/content/program-api/catalog-api.md [hunk 1]
```diff
@@ -102,6 +102,30 @@ public class DropDatabase {
 }
 ```
 
+## Alter Database
```
Comment (JingsongLi) [inline docs/content/program-api/catalog-api.md:105]: Also modify Flink SQL doc and Spark SQL doc

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java [hunk 2]
```diff
@@ -1223,13 +1226,20 @@ public static Identifier toIdentifier(ObjectPath path) {
...
     @Override
     public final void alterDatabase(
             String name, CatalogDatabase newDatabase, boolean ignoreIfNotExists)
-            throws CatalogException {
-        throw new UnsupportedOperationException();
+            throws CatalogException, DatabaseNotExistException {
+        try {
+            Database oldDatabase = catalog.getDatabase(name);
+            List<PropertyChange> changes =
+                    getPropertyChanges(oldDatabase.options(), newDatabase.getProperties());
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:1236]: Maybe we can support comment for Flink now.

File: docs/content/flink/sql-alter.md [hunk 3]
```diff
@@ -227,3 +227,11 @@ The following SQL modifies the watermark strategy to `ts - INTERVAL '2' HOUR`.
 ```sql
 ALTER TABLE my_table MODIFY WATERMARK FOR ts AS ts - INTERVAL '2' HOUR
 ```
+
+# ALTER DATABASE
+
+The following SQL sets one or more properties in the specified database. If a particular property is already set in the database, override the old value with the new one.
+
+```sql
+ALTER DATABASE [catalog_name.]db_name SET (key1=val1, key2=val2, ...)
```
Comment (JingsongLi) [inline docs/content/flink/sql-alter.md:236]: Add example to alter comment or location.

File: docs/content/spark/sql-alter.md [hunk 4]
```diff
@@ -240,3 +240,13 @@ The following SQL changes the type of a nested column `f2` to `BIGINT` in a stru
...
 ALTER TABLE my_table ALTER COLUMN v.value.f2 TYPE BIGINT;
 ```
+
+
+# ALTER DATABASE
+
+The following SQL sets one or more properties in the specified database. If a particular property is already set in the database, override the old value with the new one.
+
+```sql
+ALTER { DATABASE | SCHEMA | NAMESPACE } my_database
```
Comment (JingsongLi) [inline docs/content/spark/sql-alter.md:250]: Add example to alter comment or location.

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java (hunk 2 above)
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:1236]: Here should extract comment from newDatabase?

File: paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkCatalog.java [hunk 5]
```diff
@@ -516,10 +521,35 @@ protected List<String> convertPartitionTransforms(Transform[] transforms) {
         return partitionColNames;
     }
 
-    // --------------------- unsupported methods ----------------------------
-
     @Override
-    public void alterNamespace(String[] namespace, NamespaceChange... changes) {
-        throw new UnsupportedOperationException("Alter namespace in Spark is not supported yet.");
+    public void alterNamespace(String[] namespace, NamespaceChange... changes)
```
Comment (JingsongLi) [inline paimon-spark/paimon-spark-common/src/main/java/org/apache/paimon/spark/SparkCatalog.java:525]: Add test in `DDLWithHiveCatalogTestBase`?

File: docs/content/flink/sql-alter.md (hunk 3 above)
Comment (jerry-024) [reply docs/content/flink/sql-alter.md:236]: As in flink sql couldn't alter comment for database, so we only could alter location.

**Comment by JingsongLi** [comment]
+1

**Pull request merged**